test/server/server_test.cc

```diff
@@ -378,5 +378,19 @@ TEST_P(ServerInstanceImplTest, MutexContentionEnabled) {
   EXPECT_NO_THROW(initialize(std::string()));
 }
 
+TEST_P(ServerInstanceImplTest, NoHttpTracing) {
+  options_.service_cluster_name_ = "some_cluster_name";
+  options_.service_node_name_ = "some_node_name";
+  EXPECT_NO_THROW(initialize("test/server/empty_bootstrap.yaml"));
+  EXPECT_NE(nullptr, dynamic_cast<Tracing::HttpNullTracer*>(&server_->httpContext().tracer()));
+}
+
+TEST_P(ServerInstanceImplTest, ZipkinHttpTracingEnabled) {
+  options_.service_cluster_name_ = "some_cluster_name";
+  options_.service_node_name_ = "some_node_name";
+  EXPECT_NO_THROW(initialize("test/server/zipkin_tracing.yaml"));
+  EXPECT_EQ(nullptr, dynamic_cast<Tracing::HttpNullTracer*>(&server_->httpContext().tracer()));
```

Comment:
It might be a slightly better test to check for the Zipkin tracer type?

Reply:
Agreed...I probably should have looked deeper. Is the zipkin tracer object exposed?

In any case I could follow up with an edit tomorrow if you want to get the fix in sooner.

Reply:
It doesn't seem like there's a specific Zipkin class for this but I added a test for casting to HttpTracerImpl* in #5250.